lms/djangoapps/verify_student/models.py

```diff
@@ -887,39 +892,46 @@ def verification_status_for_user(cls, user, course_id, user_enrollment_mode):
 
 
 class VerificationCheckpoint(models.Model):
-    """Represents a point at which a user is challenged to reverify his or her identity.
-        Each checkpoint is uniquely identified by a (course_id, checkpoint_name) tuple.
     """
-
-    CHECKPOINT_CHOICES = (
-        ("midterm", "midterm"),
-        ("final", "final"),
-    )
-
+    Represents a point at which a user is asked to re-verify his/her identity.
+    Each checkpoint is uniquely identified by a
+    (course_id, checkpoint_location) tuple.
+    """
     course_id = CourseKeyField(max_length=255, db_index=True)
-    checkpoint_name = models.CharField(max_length=32, choices=CHECKPOINT_CHOICES)
+    checkpoint_location = models.CharField(max_length=255, null=True, blank=True)
     photo_verification = models.ManyToManyField(SoftwareSecurePhotoVerification)
 
     class Meta:  # pylint: disable=missing-docstring, old-style-class
-        unique_together = (('course_id', 'checkpoint_name'),)
+        unique_together = (('course_id', 'checkpoint_location'),)
 
     def __unicode__(self):
-        """Unicode representation of the checkpoint. """
+        """
+        Unicode representation of the checkpoint.
+        """
         return u"{checkpoint} in {course}".format(
             checkpoint=self.checkpoint_name,
             course=self.course_id
         )
 
+    @lazy
+    def checkpoint_name(self):
+        """
+        Lazy method for getting checkpoint name of reverification block.
+        """
+        checkpoint_key = UsageKey.from_string(self.checkpoint_location)
+        return modulestore().get_item(checkpoint_key).related_assessment
```

Comment:
@wedaly I have added exception `ItemNotFoundError` handling here. Now we will log `warning` that a checkpoint with given location not found and return  `checkpoint_location` to user instead of `related_assessment`.